auth-jwt/Sources/App/Application+configure.swift

```diff
@@ -0,0 +1,29 @@
+import Hummingbird
+import HummingbirdAuth
+import HummingbirdFoundation
+
+extension HBApplication {
+  public func configure() throws {
+    let env = HBEnvironment.shared
+    self.encoder = JSONEncoder()
+    self.decoder = JSONDecoder()
+
+    self.middleware.add(HBLogRequestsMiddleware(.debug))
+    self.middleware.add(
+      HBCORSMiddleware(
+        allowOrigin: .originBased,
+        allowHeaders: ["Accept", "Authorization", "Content-Type", "Origin"],
+        allowMethods: [.GET, .OPTIONS]
+      ))
+
+    guard let jwksUrl = env.get("JWKS_URL") else { preconditionFailure("jwks config missing") }
+    self.middleware.add(
+      BearerAuthenticator(
+        jwksUrl: jwksUrl
+      ))
+
+    router.get("/") { _ in
+      return "Hello"
```

Comment:
See comments in authenticator. Hummingbird should really have an equivalent to Vapor's `GuardMiddleware` to return 404s when a request is not authenticated. In the meantime you can add at the top of the route
```swift
guard let authenticateResponse = request.authGet(TokenPayload.self) else { 
    throw HBHTTPError(.unauthorized) 
}
```

Reply:
The middleware requires authentication for all routes, so any request without valid authorization header will return 404 at the moment

Reply:
You can use groups to apply middleware to a set of routes. 
```swift
router.group()
    .add(MyMiddleware())
    .get("/") { _ in
        return "Hello"
    }
}
```